crates/model/src/order.rs

```diff
@@ -614,6 +614,8 @@ pub struct OrderMetadata {
     /// Does not take partial fill into account.
     #[serde(default, with = "u256_decimal")]
     pub solver_fee: U256,
+    #[serde(skip)]
+    pub executed_solver_fee: Option<U256>,
```

Comment:
This feels like a code-smell that we need to read the `executed_solver_fee` into `OrderMetadata`. I don't think either the FE nor the `solver` need this field, so I don't think we should load it here.

Can we instead change the `autopilot` code to only read the order data it needs for storing the settlement observation?

Reply:
We already discussed this a bit [here](https://github.com/cowprotocol/services/pull/1303#discussion_r1124102493).

Reply:
My point is that (AFAICT) the solver observation code doesn't need to load full `Order`s from the database, it can have a custom order type and avoid this.

Reply:
Totally. I think we all agree on that.

Reply:
I removed the `model::Order` dependency and now `autopilot` reads directly from `FullOrder`.